packages/shared/src/components/Feed.spec.tsx

```diff
@@ -1058,3 +1072,108 @@ it('should report irrelevant tags', async () => {
     ).not.toBeInTheDocument(),
   );
 });
+
+describe('acquisition form card', () => {
+  const replaceRouter = jest.fn();
+  const url = new URL(`http://localhost:5002?${acquisitionKey}=true`);
+  const mockedQuery = { [acquisitionKey]: 'true' };
+
+  beforeEach(() => {
+    /* eslint-disable @typescript-eslint/no-var-requires,global-require */
+    mockedQuery[acquisitionKey] = 'true';
+    mocked(useRouter).mockImplementation(() => ({
+      route: '/',
+      pathname: '',
+      query: mockedQuery,
+      asPath: '',
+      push: jest.fn(),
+      events: {
+        on: jest.fn(),
+        off: jest.fn(),
+      },
+      replace: replaceRouter,
+      beforePopState: jest.fn(() => null),
+      prefetch: jest.fn(() => null),
+    }));
+
+    Object.defineProperty(window, 'location', {
+      value: url,
+      configurable: true,
+    });
+  });
+
+  it('should show the form card when the right url query param is present', async () => {
+    renderComponent([createFeedMock()], defaultUser);
+    await screen.findByTestId('acquisitionFormCard');
+  });
+
+  it('should hide the card once form is submitted', async () => {
+    let mutationCalled = false;
+    const { unmount } = renderComponent([createFeedMock()], defaultUser);
+
+    const card = await screen.findByTestId('acquisitionFormCard');
+    expect(card).toBeInTheDocument();
+
+    const radio = await screen.findByLabelText('Other');
+    fireEvent.click(radio);
+
+    mockGraphQL({
+      request: {
+        query: USER_ACQUISITION_MUTATION,
+        variables: { acquisitionChannel: 'other' },
+      },
+      result: () => {
+        mutationCalled = true;
+        return { data: { addUserAcquisitionChannel: { _: null } } };
+      },
+    });
+
+    const submit = await screen.findByText('Submit');
+    fireEvent.click(submit);
+
+    await waitFor(() => {
+      expect(mutationCalled).toBeTruthy();
+    });
+
+    // set mocked query to false
+    mockedQuery[acquisitionKey] = 'false';
+
+    // rerender
+    unmount();
+    renderComponent([createFeedMock()], defaultUser);
```

Comment:
More of a note, we do have a actual re-render if you need:
https://github.com/dailydotdev/apps/blob/ea73e392446905fba18348b9d5629418b24f04ec/packages/shared/src/contexts/AnalyticsContext.spec.tsx#L270

Reply:
Yeah, but that expects a react element to be passed in. So I would need to create another function that returns the component. Seems easier this way :)